### DISCUSSION TITLE
Could I use sub-navigation in filament-page?

### DISCUSSION THREAD

**Discussion by devhoussam1998** [filamentphp/filament#10073] [Help]
https://filamentphp.com/docs/3.x/panels/resources/getting-started#resource-sub-navigation

**Comment by uadoronin** [comment]
This would be a nice feature for the Settings page (for example). 
Unfortunately, there is no documentation of use Sub Navigation on the Filament Pages.

**Reply by devhoussam1998** [reply]
yeah it will be nice to have sub-navigation for pages i hope the core team think about it in future releases.

**Comment by arnaudlidz** [comment]
Definitively

**Comment by StaffCollab** [comment]
Totally agree.

This is a much needed feature for custom pages

**Reply by devhoussam1998** [reply]
yeah I'm working on project now and totally needed

**Comment by divdax** [comment]
Maybe try this: https://filamentphp.com/plugins/aymanalhattami-page-with-sidebar

**Reply by devhoussam1998** [reply]
I know about this package but it's better to be in the Core rather than using plugins.

**Comment by HelenaGwyn** [comment]
You can do this:

```php
public function getSubNavigation(): array
{
    return $this->generateNavigationItems([
        Page::class,
        Page::class,
        ...
    ]);
}
``` 

This will take all the navigation settings of the Page as used by getNavigationItems() of the Page-class.

Or you can do this:

```php
public function getSubNavigation(): array
{
    return [
            NavigationGroup::make('group')
                ->items([
                    NavigationItem::make(Page::class::getNavigationLabel())
                        ->icon(Pages::class::getNavigationIcon())
                        ->isActiveWhen(fn (): bool => request()->routeIs(Page::class::getNavigationItemActiveRoutePattern()))
                        ->url(Page::class::getNavigationUrl()),
                ])
        ];
}
``` 

to build your own custom subnavigation with the settings you want.

**Reply by rosfandy** [reply]
Thank you, the second one solved
